### ISSUE TITLE
TPE sampler on pruned trials

### ISSUE THREAD

**Issue by fsmosca** [optuna/optuna#2895]
Does TPE sampler considers pruned trials on its algorithm?

If the trial is pruned on param_a=20 because of bad objective value, does it suggest again in future trials the param_a=20?

**Comment by nzw0301** [comment]
Yes. Please see this issue https://github.com/optuna/optuna/issues/1647 for more details. I think it depends on the other hyper-parameters.
